widget/gonk/nsWindow.cpp

```diff
@@ -134,6 +134,20 @@ nsWindow::ConfigureAPZControllerThread()
     APZThreadUtils::SetControllerThread(CompositorBridgeParent::CompositorLoop());
 }
 
+void
+nsWindow::ForceComposition()
+{
+    // The only implementation of nsIWidget::ForceComposition in nsWindow
+    // is for remote control.
+    if (gFocusedWindow) {
+        // NB: this is a racy use of gFocusedWindow.  We assume that
+        // our one and only top widget is already in a stable state by
+        // the time we start receiving mousemove events.
+        gFocusedWindow->mCompositorBridgeParent->InvalidateOnCompositorThread();
+        gFocusedWindow->mCompositorBridgeParent->ScheduleRenderOnCompositorThread();
+    }
+}
+
```

Comment:
What has been done here (I'd prefer the name InvalidateAndScheduleRender which is more precise.) is pretty much the same as |nsWindow::NotifyHoverMove(const ScreenIntPoint& point)|.  Could you help merge these two functions or let this function be called in |nsWindow::NotifyHoverMove| ?